### PR TITLE
Feat: fuel management

### PR DESCRIPTION
**Pull request by dleard**

### CODE REVIEW

File: ggircs-app/app/components/FuelManagement/MappedFuelTypeTable.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          }
+        }
+      }
+    `,
+    normalizedFuelType
+  );
+
+  const [deleteFuelMapping] = useDeleteFuelMappingMutation();
+
+  const handleRemove = (id: string) => {
```
Comment (matthieu-foucault) [inline ggircs-app/app/components/FuelManagement/MappedFuelTypeTable.tsx:37]: After deleting a fuel mapping, I was expecting to see it in the "Unmapped Fuels" list, but didn't

File: ggircs-app/app/components/FuelManagement/MappedFuelTypeTable.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      </Alert>
+    );
+  }
+
+  return (
+    <>
+      <Table striped bordered hover>
+        <thead>
+          <tr>
+            <th scope="col">Mapped Fuel Type Variations</th>
```
Comment (matthieu-foucault) [inline ggircs-app/app/components/FuelManagement/MappedFuelTypeTable.tsx:67]: Should there be a help tooltip indicating that this is the fuel name found in the SWRS reports?

**Review by matthieu-foucault — COMMENTED** [review]
Found an issue when removing fuel mappings.

Should this UI have a button to create a new Normalized Fuel? Do we have another issue for that?

File: deploy/swrs/parameters/queries/unmapped_fuel.sql [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    join swrs.report
+    on fuel.report_id = report.id
+    left join ggircs_parameters.fuel_mapping fm
+    on fuel.fuel_mapping_id = fm.id
+    where
+      (
+        fuel.fuel_type not in (select fuel_type from ggircs_parameters.fuel_mapping)
+        or fm.fuel_carbon_tax_detail_id is null
+      )
+      and report.reporting_period_duration > 2014;
```
[the remark this answers is not in the document]
Comment (dleard) [reply deploy/swrs/parameters/queries/unmapped_fuel.sql:21]: This is how the metabase alert is written, there is an old fuel that needs to be ignored. Though we could just ignore fuel_type='Other' here instead.

File: deploy/swrs/parameters/queries/unmapped_fuel.sql [hunk 4]
```diff
@@ -0,0 +1,26 @@
+-- Deploy ggircs:swrs/parameters/queries/unmapped_fuel to pg
+-- requires: swrs/parameters/schema
+-- requires: swrs/parameters/table/fuel_mapping
+-- requires: swrs/public/table/fuel
+
+begin;
+
+create or replace function ggircs_parameters.unmapped_fuel()
+  returns table (fuel_type text, fuel_mapping_id int)
```
[the remark this answers is not in the document]
Comment (dleard) [reply deploy/swrs/parameters/queries/unmapped_fuel.sql:9]: No reason, I can add a custom type & use that

**Comment by dleard** [comment]
> Found an issue when removing fuel mappings.
> 
> Should this UI have a button to create a new Normalized Fuel? Do we have another issue for that?

We have this in a separate issue #394

File: ggircs-app/app/components/FuelManagement/MappedFuelTypeTable.tsx (hunk 1 above)
Comment (dleard) [reply ggircs-app/app/components/FuelManagement/MappedFuelTypeTable.tsx:37]: Deleting a fuel mapping will show up in the "Unmapped Fuels" list only if the fuel exists in the fuel table. There will be no data in swrs.fuel locally unless you add it.

File: deploy/swrs/parameters/queries/unmapped_fuel.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (matthieu-foucault) [reply deploy/swrs/parameters/queries/unmapped_fuel.sql:21]: 'Other' would be better. Some `2014` reports may still be submitted nowadays

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**